server/src/main/java/org/apache/kafka/server/AssignmentsManager.java

```diff
@@ -202,9 +199,12 @@ public void run() throws Exception {
                     return;
                 }
                 if (existing.timestampNs > timestampNs) {
-                    existing.onComplete();
-                    if (log.isDebugEnabled()) log.debug("Dropping assignment {} because it's older than {}", this, existing);
+                    existing.merge(this);
+                    if (log.isDebugEnabled()) log.debug("Dropping assignment {} because it's older than existing {}", this, existing);
```

Comment:
I think we can avoid `log.isDebugEnabled()` and the like when using parameterised messages. Refer https://www.slf4j.org/faq.html#logging_performance